src/time_series_parser.jl

```diff
@@ -159,12 +171,15 @@ function handle_normalization_factor(
     return data
 end
 
+get_max_value(ta::TimeSeries.TimeArray) = maximum(TimeSeries.values(ta))
+get_max_value(ta::Vector) = maximum(ta)
+
 function handle_normalization_factor(
-    ta::TimeSeries.TimeArray,
+    ta::Union{TimeSeries.AbstractTimeSeries, AbstractArray},
```

Comment:
I think that we should implement this function on a TimeSeriesData type.  The original function returns a new TimeArray because TimeArrays are immutable.  In the case of Deterministic we can modify the data in place.

Reply:
Should we handle this in a later PR? It seems like an effort on its own.

Reply:
Agreed. Can you make a case for tracking?